subgit/core.py

```diff
@@ -488,11 +488,13 @@ def pull(self, names):
                 log.info(f"Current commit summary on HEAD in git repo '{name}': ")
                 log.info(f"  {str(repo.head.commit.summary)}")
 
-    def delete(self, repo_names=None):
+    def delete_reset(self, repo_names=None, command=None):
```

Comment:
Okay, so I thought about how to rewrite it, and although possible, maybe a bit redundant. So for the time being I agree with it being a bit cleaner to split them and not have them colide.

Reply:
So one way to rewrite this to avoid having this method being a dispatcher is to rewrite it and remove the dispatching part but only do the validation and question part as that is what is shared and then you call this method from each of the command methods so in the beginning of the _reset and _clean methods you call this helper to perform the config extraction, ask the question and validation, so in that case we can still call the separate methods and avoid any dispatch method